### DISCUSSION TITLE
Any way to enable Requests from all songs, not just the current playlist?

### DISCUSSION THREAD

**Discussion by BillDewey** [AzuraCast/AzuraCast#7334] [Q&A]
I often want listeners to request anything they would like to here, regardless of any criteria.  I can't find a way to do this, all my music is in the Default Playlist, if I Disable this list so AutoDj does not pull from it, then anything not on the currently enabled list is not available.  Hopefully I am just missing some combination of settings.  

I have also noticed that there seems to be no way to start AutoDJ even with a single playlist enabled, and have it not auto-populate the Song Cue unless I set the Manual AutoDJ in the settings which turns off the Queue.  If I am missing something along the way, please point me to the right part of the documents so I can understand better.

**Comment by Vaalyn** [comment]
You can make a playlist of type `Advanced` so it's not included in any AutoDJ scheduling, mark it as allow requests from it and then put all your media directories into that playlist so that the playlist includes everything. That's how I have it setup on my station for stuff that should be requestable but not automatically included in playout from that playlist.

**Reply by BillDewey** [reply]
Thank you so very much, this works great.  It is also nice that I can just add the "requestable" folders as I need them.  I was worried about trying the "Advanced" as it talks about needing to build Liquidsoap configurations, and I am not anywhere near attempting that.

Am I on the right track with using Manual AutoDJ to better control the playlist without the Queue?  With these two I think I have solved my main issues.  I am trying to migrate from an Everestcast implementation, and I do not manage the Azuracast install, so need to do what I need through the Interface, which seems very robust.

You folks are doing a great job in my opinion.

**Reply by Vaalyn** [reply]
I personally wouldn't recommend the manual mode as that will put all scheduling completely on Liquidsoap, making it necessary to restart LS on each change that you want it to apply. For example if you want to add a new playlist to the rotation or want to change the schedule of one you'll have to restart LS.

It would be better to look into the logs (application log of the installation & liquidsoap log of the station) to check why the queue isn't populating with manual mode off and get that figured out since it will be less of a headache to manage than manual mode imho. My perspective of the manual mode is that it's more for people who are very familiar with LS coding and want to build their scheduling completely by hand.

With manual mode I personally find it pretty hard to help debug stuff since it often times is more of a black box to me than the AC AutoDJ based on the queue.

**Reply by BillDewey** [reply]
Thank you again.  For background, I am coming from an Everestcast system where starting a playlist that was Sequential would always start with the first song.  My situation is that at times I follow another DJ who may not always end right at a scheduled time, may be a minute or two over or under, and I need to ensure that my playlist starts "on command" playing from the first song in a sequential list.  

When not in Manual Mode it is not an issue of the Queue not populating, more that it does not populate with the first songs in the Playlist.  I am trying to do this by Disabling the list, then Enabling it as the DJ before me ends.  What I found when not in Manual mode, was that the Queue always seems to be populated but not starting at the first song.  I tried to clear the Queue and Restart Broadcasting, but even then the Queue would populate first and not play my list from the beginning.  

This is more difficult to explain than I thought.  The other thing I noticed is that Requests seem to play sooner in Manual Mode, which now makes perfect sense, as the system does not wait for the already populated Queue.

If there is a way to always force a sequential list to play from the first song every time, with Manual Mode off, that would be ideal.  I will do some more testing to see, if not then for my situation Manual may be the only way to manage this.

If you have pointers to other reading or videos, would help as well.  This interface has so many really great things about it, I just need to get through the learning curve of making it work for my situation.  

One thing that might work for me is to have one station in Manual Mode for the On-Demand lists and another that runs more "normally" for scheduled lists.  Would that make sense?

More testing to do, let me know if there is more information that would help.  Thank you, and the team, for all the assistance.  As I do not administer the installation, if there is something you need from a system log I can ask for it.

**Reply by Vaalyn** [reply]
I'm not familiar with how EverestCast works so I can't really compare it directly.

> My situation is that at times I follow another DJ who may not always end right at a scheduled time, may be a minute or two over or under, and I need to ensure that my playlist starts "on command" playing from the first song in a sequential list.

Not sure exactly what you mean with "on command" though if you have a scheduled sequential playlist the AzuraCast AutoDJ will check if the playlists internal queue is completely filled at the moment it should start, checks if it has played something from it in the scheduled timeframe and if it detects that the playlists internal queue is not full (in other words if it is not at the beginning of the sequential order) it will reset it so that it should begin at the first song in the order of the sequential playlist.

This is however only for scheduled playlists since unscheduled (general rotation or other types without schedule entry) ones have no pre-defined "start point" with which we could determine when a reset would be necessary.

> When not in Manual Mode it is not an issue of the Queue not populating, more that it does not populate with the first songs in the Playlist. I am trying to do this by Disabling the list, then Enabling it as the DJ before me ends. What I found when not in Manual mode, was that the Queue always seems to be populated but not starting at the first song. I tried to clear the Queue and Restart Broadcasting, but even then the Queue would populate first and not play my list from the beginning.

From the description this sounds like you have an unscheduled sequential playlist that you'd like to have reset after a DJ was live. I think that's currently not implemented as a feature, might be possible to do though. You could add it as a feature request over at https://azuracast.featureupvote.com/ though.

As a work around that doesn't require restarting you could use the "Reshuffle" button on the sequential playlist here and then clear the upcoming song queue:
<img width="1311" alt="image" src="https://github.com/user-attachments/assets/744be3a8-3f96-439c-88a3-93f43aa34b33">
The naming might be a bit misleading in the specific case of sequential playlist since you don't want to "shuffle" the playlist but all this function does is reset the internal queue of the playlist which in the case of sequential ones means reset to the beginning of the defined order 😄 

> If you have pointers to other reading or videos, would help as well. This interface has so many really great things about it, I just need to get through the learning curve of making it work for my situation.

I don't really know about the current state of tutorial videos on YT or other places since I personally haven't needed any on account of having worked with the code for years now, maybe someone else knows more about this 😅 

> One thing that might work for me is to have one station in Manual Mode for the On-Demand lists and another that runs more "normally" for scheduled lists. Would that make sense?

Might work although in my experience having remote playlists / streams can introduce it's own set of unique challenges and issues. Maybe an Advanced type playlist with custom Liquidsoap code for it could work. Would be a bit of a steeper learning curve though since LS can be quite complex depending on what needs to be achieved exactly and how it should interop with the existing structure of the LS config. Here a link to the LS docs for the currently used version: https://www.liquidsoap.info/doc-2.2.5/reference.html

**Comment by BillDewey** [comment]
Everestcast has a "start playlist" button that actually schedules the playlist to start at the beginning of the next minute. I can have a "filler" as my first song, start my list a minute before the previous DJ ends, and then "skip track" as soon as I hear my filler. The Playlist always starts from the first song.

It looks like your method will work as long as I schedule it manually, reshuffle, clear the queue.  I just have to figure out the right sequence and the timing with the last DJ.  What appears to happen is that when the scheduled time hit, the playlist did not start until the last playing song finished.  Given the way the Queue works with AutoDJ this makes sense, it just isn't quite what I need.

I am not sure which is more difficult timing wise.  With my work around with Manual, I can control things better as I don't have the Queue to contend with.  And since I don't run a "scheduled" station the other issues don't matter as much, such as restarting as I have no "listeners".

A bit more testing and I think I can make this work for me for now.

As I think more about this, a Playlist of type "Play on Demand" that tells the AutoDJ to clear the queue and start the playlist now is what would be ideal.  If this makes sense to you, this would be much like combing your Reshuffle and Clear Queue but would be much faster 
and seamless, I would be most happy to post it as a feature request.

I looked at the LiquidSoap bits, and, well, after 50+ years of Software Development myself, I am too old and lazy now, that is too much like work.

Thanks again, this is helping me understand the system much better, I know that I am not doing things the way a "normal" station does, many people I know just run "live" with something like VirtualDJ or MIXXX on their PC.  I will look around on Youtube and see if I can find anything of this nature as well.
